### PR TITLE
Favor the function over the method syntax when cloning a reference counted pointer.

### PR DESCRIPTION
**Pull request by nical**
```data.clone()``` reads like performance hazard, due to the vague definition of Clone, which can be either very expensive if data is an uniquely owned large resource or very cheap if it is a reference counted pointer to such a resource. This has already confused people reading the code and even code reviews several times in webrender, so I would like to work around this unfortunate ambiguity.

I first tried to address this at the [standard library level](https://github.com/rust-lang/rfcs/pull/1954) but the decision of the rust team appears to be that we should use the function syntax instead of the method syntax to address the problem.

See also the [clippy issue](https://github.com/Manishearth/rust-clippy/issues/1645) about providing a lint for this.

I decided to take a systematic approach and use the function call syntax for all of the outstanding reference counted pointer clones that I came across (I would not be surprised that I missed some of them). Rather than try to separate the obvious from the ambiguous ones. I would rather make this a default policy than have to debate about where the line is in each code review.

The three commits are independent and can be reviewed separately (although they are small and simple enough to look at as one diff if you prefer).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/1037)

### CODE REVIEW

File: webrender/src/resource_cache.rs [hunk 1]
```diff
@@ -709,7 +709,7 @@ impl ResourceCache {
                         is_opaque: image_descriptor.is_opaque,
                     }
                 } else {
-                    image_template.descriptor.clone()
+                    image_template.descriptor
```
Comment (nox) [inline webrender/src/resource_cache.rs]: This change is not related to what the PR describes, please remove of put in its own commit.

Comment (kvark) [inline webrender/src/resource_cache.rs]: @nox I think you are overly sensitive here. It doesn't appear to me that the change bears any danger, and it's a bit related to the `clone()` refactor.

**Comment by kvark** [comment]
@nical I see where you are coming from, and the change looks clean.
However, I'm not yet convinced that this is the way to go for all Rust libraries. The RFC discussion basically concluded that if you want it, you are free to use it, and the clippy issue discussion is empty so far (I know, the issue is recent). I'd like to get some opinion from @glennw  and Rust Core team.

**Comment by glennw** [comment]
I really like this idea.

**Comment by nical** [comment]
I did a last rebase before I disappear for 2 weeks. Don't hesitate to re-submit this in my place if a consensus is approached while I'm away.
I really think we would benefit from favoring this syntax. The rust core team decided stand on the side of letting people use the function syntax if they want to. Whether we want to hand pick places where we use it or apply it consistently is really up to us. Clippy support would be a plus but doesn't affect whether this style is to be considered good or bad for us.

**Comment by kvark** [comment]
I'm not going to block this. @glennw please proceed if you are happy about it.

**Comment by glennw** [comment]
OK, let's merge it. We can always change back if we decide it's not worthwhile. Thanks!
